### DISCUSSION TITLE
url 이미지 읽어오기 관련 이슈

### DISCUSSION THREAD

**Discussion by HuttTheJAVA** [100-hours-a-week/discussions#9] [개발] [answered]
특정 url로 연결되는 이미지를 받아서 S3로 업로드하던 코드가 이미지 생성 ai를 바꾸면서(OpenAi -> Karlo) 동작하지 않습니다. <br>
흐름은 다음과 같습니다. <br><br>url이 매개변수로 들어온다 -> url의 이미지를 File 타입으로 받아온다 -> S3 서버에 File을 업로드한다. <br>
이때 url을 File로 저장하는 아래 사진의 메소드에서 ImageIO.read부분에서 코드가 실행을 중단하고(ide에 에러 메시지 출력 X) 그냥 끝나버립니다. <br>포스트맨으로 확인해 본 결과 "Can't get input stream from URL!"가 출력됩니다.. 코드 문제인가 해서 ai 생성 이미지 말고 아무 이미지의 url로 코드를 실행시켜보니 s3에 잘 업로드 됩니다. <br> 어떤 점이 문제일까요..? 제프의 의견을 여쭙고 싶습니다!<br><br>
<img width="826" alt="image" src="https://github.com/user-attachments/assets/d2523806-8a8f-4d5f-934f-97323e90c4bb">

**Comment by ej31** [comment, marked as answer]
## 답변
- 칼로 문서를 확인해보니 응답 중 이미지 관련된 내용은 `images` 키 안에 포함 되어 있는 것으로 보입니다.
<img width="864" alt="image" src="https://github.com/user-attachments/assets/e0fdee1b-5b23-4469-b800-116e6da86860">
 - 원문: https://developers.kakao.com/docs/latest/ko/karlo/rest-api#text-to-image-response-body

- 지금 코드를 보면 ImageIO 는 URL을 통해 돌아온 응답 자체가 이미지 스트림이어야 합니다.
- 또한 문서에서 보면 Image URL 을 줄 수도 있고 base64 를 줄 수도 있다고 하니 이 부분은 분기 처리 하면 되겠습니다.

**Reply by HuttTheJAVA** [reply]
답변 감사합니다!! 칼로 API에서 wepb로 응답오는 이미지를 jpg로 변환하여 해결하였습니다.
